test_api.py

```diff
@@ -0,0 +1,128 @@
+import requests
+import sqlite3
+
+from datetime import datetime
+# from pprint import pprint as pp
+
+
+DB = 'test.db'
+ID_G = iter(range(1, 10_000))
+
+init_data = [
+    (next(ID_G), 'Prepare a dinner for a week', 'home',
+     datetime.today().strftime('%d-%m-%y'), 0, ''),
+    (next(ID_G), 'Make the report of the week', 'job',
+     datetime.today().replace(day=datetime.today().day + 2).strftime('%d-%m-%y'),
+     0, ''),
+    (next(ID_G), 'Make a homework for lesson 10', 'studies',
+     datetime.today().replace(day=datetime.today().day + 1).strftime('%d-%m-%y'),
+     0, ''),
+    (next(ID_G), 'Read the documentation about descriptors', 'studies',
+     datetime.today().strftime('%d-%m-%y'), 0, '')
+]
+
+
+def test_add_new_task():
+    print('test POST')
+    new_task = {
+        'id': next(ID_G),
+        'title': 'Finish task 4',
+        'category': 'job',
+        'expired_date': datetime.today().replace(day=datetime.today().day + 2).strftime('%d-%m-%y'),
+    }
+
+    response = requests.post('http://localhost:5000/api/tasks/', json=new_task)
+    # print(response.content)
+    assert response.status_code == 201
+    response = requests.get(f'http://localhost:5000/api/tasks/'
+                            f'?id={new_task["id"]}')
```

Comment:
Why do you use `?id` like HTTP parameter? But not part of your URL via a [variable](https://flask.palletsprojects.com/en/1.1.x/quickstart/#variable-rules)?

Reply:
Really don't know :) May be it was an idea try to use more "broad-coverage' handler for `/api/tasks/` path. But you are right, in this case there was already implemented more explicit handler, which is more siutable here. Thank you.